herddb-core/src/main/java/herddb/core/TableSpaceManager.java

```diff
@@ -514,6 +522,49 @@ void apply(CommitLogResult position, LogEntry entry, boolean recovery) throws Da
                 writeTablesOnDataStorageManager(position, false);
             }
             break;
+            case LogEntryType.TABLE_CONSISTENCY_CHECK: {
+                try {
+                    TableChecksum check = new ObjectMapper().readValue(entry.value.to_array(), TableChecksum.class);
+                    String tableSpace = check.getTableSpaceName();
+                    String query = check.getQuery();
+                    String tableName = entry.tableName;
+                    //In the entry type = 14, the follower will have to run the query on the transaction log
+                    if (!isLeader()) {
+                        AbstractTableManager tablemanager = this.getTableManager(tableName);
+                        DBManager manager = this.getDbmanager();
+
+                        if (tablemanager == null || tablemanager.getCreatedInTransaction() > 0) {
+                            throw new TableDoesNotExistException(String.format("Table %s does not exist.", tablemanager));
+                        }
+                        /*
+                            scan = true
+                            allowCache = false
+                            returnValues = false
+                            maxRows = -1
+                        */
+                        TranslatedQuery translated = manager.getPlanner().translate(tableSpace, query, Collections.emptyList(), true, false, false, -1);
+                        TableChecksum scanResult = TableDataChecksum.createChecksum(manager, translated, this, tableSpace, tableName);
+                        long followerDigest = scanResult.getDigest();
+                        long leaderDigest = check.getDigest();
+                        long leaderNumRecords = check.getNumRecords();
+                        long follNumRecords = scanResult.getNumRecords();
+                        long table_scan_duration = check.getScanDuration();
```

Comment:
What is this? This is the value written on the entry. It does not have value here